Pull in r203624 from upstream clang trunk (by Hans Wennborg):

  Be case sensitive when determining driver mode based on argv[0] except on Windows

  This narrows the impact of r188833 after Dimitry pointed out that it's good to be
  able to tell the difference between 'cc' and 'CC'.

Introduced here: http://svnweb.freebsd.org/changeset/base/263048

Index: tools/clang/tools/driver/driver.cpp
===================================================================
--- tools/clang/tools/driver/driver.cpp
+++ tools/clang/tools/driver/driver.cpp
@@ -229,8 +229,10 @@ static void ParseProgName(SmallVectorImpl<const ch
     { "++",        "--driver-mode=g++" },
   };
   std::string ProgName(llvm::sys::path::stem(ArgVector[0]));
+#ifdef _WIN32
   std::transform(ProgName.begin(), ProgName.end(), ProgName.begin(),
                  toLowercase);
+#endif
   StringRef ProgNameRef(ProgName);
   StringRef Prefix;
 
